include/rmm/device_scalar.hpp

```diff
@@ -92,6 +92,14 @@ class device_scalar {
 
  private:
   rmm::device_buffer buff{sizeof(T)};
+
+  inline void _memcpy(void *dst, const void *src, cudaStream_t stream) const {
+    auto status = cudaMemcpyAsync(dst, src, sizeof(T), cudaMemcpyDefault,
+                                  stream);
+    if (cudaSuccess != status) {
```

Comment:
```suggestion
    status = cudaStreamSynchronize(stream);
    
    if (cudaSuccess != status) {
```

Reply:
Won't this perform two stream syncs, one implicitly via `cudaMemcpyAsync` (due to host pageable memory) and another with the explicit sync?

Reply:
Since the second sync is immediately after the first the cost should be negligible.

Reply:
Reintroduced explicit synchronization and updated documentation to reflect the explicit nature of the sync.